compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1482,7 +1482,8 @@ public void visit(BLangTupleVariable varNode) {
 
         // Create tuple destruct block stmt
         final BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(varNode.pos);
-
+        blockStmt.parent = env.node;
+        blockStmt.internal = true;
```

Comment:
I assume internal was already used in other cases and/or should be used in other cases. Did we verify and add tests for these different scenarios?